### PR TITLE
feat: Add support for gcode viewing on plotter-style devices

### PR DESCRIPTION
**Pull request by kieraneglin**
Previously, the gcode viewer did not work if no extrusion moves were defined. This broke the viewer in cases where Klipper was being used on 2d plotters without a Z or extruder.

Signed-off-by: Kieran Eglin <kieran.eglin@gmail.com>

### CODE REVIEW

File: src/store/gcodePreview/getters.ts [hunk 1]
```diff
@@ -48,21 +48,31 @@ export const getters: GetterTree<GcodePreviewState, RootState> = {
...
         zLast = z
 
         output.push({
           z,
           move: zStart,
-          filePosition: moves[index].filePosition
+          filePosition: move.filePosition
         })
       }
+    })
```
Comment (kieraneglin) [inline src/store/gcodePreview/getters.ts:66]: This is largely just noise. I refactored the `for` loop to a `forEach` but functionality remains unchanged

Comment (pedrolamas) [reply src/store/gcodePreview/getters.ts]: Can't remember what it was specifically, but I think I've had issues doing that in the past... 🤔

**Comment by pedrolamas** [comment]
Thanks for sending this PR, @kieraneglin.

I saw that you provided a quick example gcode for this on Discord, but can I ask that you upload a sample file so we can run some tests?

**Comment by kieraneglin** [comment]
@pedrolamas sure thing! I've attached a file that creates a small spiral using only XY moves as well as a reference for its expected output (courtesy of https://sandify.org/). Just rename the .txt to .gcode 🤙 

Screen Shot 2023-02-03 at 7 54 07 AM
[small_spiral_testing.txt](https://github.com/fluidd-core/fluidd/files/10580464/small_spiral_testing.txt)

**Comment by pedrolamas** [comment]
Thanks for providing that @kieraneglin, it was quite useful to test - and even allowed me to find a non-related bug on the app-slider (for which I now have a fix!)

**Review by pedrolamas — APPROVED** [review]

**Comment by pedrolamas** [comment]
Thank you @kieraneglin , can you please add a footer to the main PR comment with a "Signed-off-by" line, following the [CONTRIBUTING](https://github.com/fluidd-core/fluidd/blob/develop/CONTRIBUTING.md) guidelines (_you are missing the email address on yours!_)

Thank you!

**Comment by kieraneglin** [comment]
@pedrolamas updated!

**Review by matmen — APPROVED** [review]

**Pull request merged**